app/src/main/java/app/revanced/integrations/patches/SpoofSignatureVerificationPatch.java

```diff
@@ -1,40 +1,97 @@
+    private static final String PROTOBUF_PARAMETER_GENERAL = "CgIQBg";
 
     /**
-     * Protobuf parameter of short and YouTube story.
+     * Protobuf parameter of shorts and YouTube stories.
      */
-    private static final String SHORTS_PROTOBUF_PARAMETER = "8AEB"; // "8AEByAMTuAQP"
+    private static final String PROTOBUF_PARAMETER_SHORTS = "8AEB"; // "8AEByAMTuAQP"
 
     /**
      * Target Protobuf parameters.
      * Used by the generic player.
      */
-    private static final String TARGET_PROTOBUF_PARAMETER = "YADI";
-
-    public static String getVerificationSpoofOverride(String original) {
-        PlayerType player = PlayerType.getCurrent();
-        LogHelper.printDebug(() -> "Original protobuf parameter value: " + original + " PlayerType: " + player);
-        if (original.startsWith(TARGET_PROTOBUF_PARAMETER)  || original.length() == 0) {
-            if (player == PlayerType.INLINE_MINIMAL) {
-                return GENERAL_PROTOBUF_PARAMETER; // home feed autoplay
+    private static final String PROTOBUF_PARAMETER_TARGET = "YADI";
+
+    /**
+     * Injection point
+     *
+     * @param original original protobuf parameter
+     */
+    public static String getProtoBufParameterOverride(String original) {
+        try {
+            if (!SettingsEnum.SIGNATURE_SPOOFING.getBoolean()) {
+                return original;
             }
-            if (player.isNoneOrHidden()) {
-                return SHORTS_PROTOBUF_PARAMETER; // short or story
+            PlayerType player = PlayerType.getCurrent();
+            LogHelper.printDebug(() -> "Original protobuf parameter value: " + original + " PlayerType: " + player);
+            if (original.startsWith(PROTOBUF_PARAMETER_TARGET) || original.length() == 0) {
+                if (player == PlayerType.INLINE_MINIMAL) {
+                    return PROTOBUF_PARAMETER_GENERAL; // home feed autoplay
+                }
+                if (player.isNoneOrHidden()) {
+                    return PROTOBUF_PARAMETER_SHORTS; // short or story
+                }
+                return PROTOBUF_PARAMETER_SHORTS; // regular video player
             }
-            return SHORTS_PROTOBUF_PARAMETER; // regular video player
+        } catch (Exception ex) {
+            LogHelper.printException(() -> "getProtoBufParameterOverride failure", ex);
         }
 
         return original;
     }
+
+
+    /**
+     * Injection point. Runs off the main thread.
+     *
+     * Used to check the status of http calls made to YouTube.
+     * Calls of interest have status code in the 4xx range (usually 403),
+     * which usually indicates a signature verification failure.
+     *
+     * @param statusCode           HTTP status code of the completed YouTube connection
+     * @param urlConnectionHeaders all connection headers of the completed connection
+     */
+    public static void connectionCompleted(int statusCode, Map<String, List<String>> urlConnectionHeaders) {
```

Comment:
What is the second parameter used for?

Reply:
Since this method hooks in a quite low-level method to make requests, if an unrelated request returns a matching exception, it could falsely detect playback issues.

Reply:
It is likely that all requests are highly abstracted, so some kind of hooking would be required to determine if the request belongs to the one, we are interested in. Maybe, instead of the URL, the request made to can be used to check, if we are currently hooking the correct request.

Reply:
`UrlResponseInfo.getUrl()` can be used for that.

Reply:
In that case, no check for the URL or domain is required and can be removed.